lib/std/c.zig

```diff
@@ -39,6 +39,32 @@ pub fn versionCheck(comptime glibc_version: std.SemanticVersion) type {
     };
 }
 
+/// Compatibility function, used by Zig's implementation of libc in `lib/c.zig`.
+/// Returns `-1` if `a` appears before `b` in lexicographical order.
+/// Returns `1` if `a` appears after `b` in lexicographical order.
+/// Returns `0` if `a` and `b` are equal.
+/// See also `std.mem.order` for idiomatic function
+/// with parameters as Zig's slices.
+pub fn strcmp(a: [*:0]const u8, b: [*:0]const u8) i8 {
```

Comment:
This file is supposed to have bindings to libc only. The appropriate place for a strcmp implementation is in lib/c.zig which is the actual implementation of libc functions.

I suggest to take @ifreund's suggestion to make a string comparison function that is not C related, but is having to do with two null terminated strings, put it in mem.zig and call it from both here and from lib/c.zig. Equivalent to how strlen is handled via std.mem.len.